doc/sphinx/proof-engine/tactics.rst

```diff
@@ -4811,3 +4811,103 @@ references to automatically generated names.
    :name: Mangle Names Prefix
 
    Specifies the prefix to use when generating names.
+
+Performance-oriented tactic variants
+------------------------------------
+
+.. tacn:: change_no_check @term
+   :name: change_no_check
+
+   For advanced usage. Similar to :n:`change @term`, but as an optimization,
+   it skips checking that :n:`@term` is convertible to the goal.
+
+   Recall the Coq kernel typechecks proofs again when they are concluded to
+   ensure safety. Hence, using :tacn:`change` checks convertibility twice
+   overall, while :tacn:`convert_concl_no_check` can produce ill-typed terms,
+   but checks convertibility only once.
+   Hence, :tacn:`convert_concl_no_check` can be useful to speed up certain proof
+   scripts, especially if one knows by construction that the argument is
+   indeed convertible to the goal.
+
+   In the following example, :tacn:`convert_concl_no_check` replaces :g:`False` by
+   :g:`True`, but :g:`Qed` then rejects the proof, ensuring consistency.
+
+   .. example::
+
+      .. coqtop:: all abort
+
+         Goal False.
+           change_no_check True.
+           exact I.
+         Fail Qed.
+
+   :tacn:`change_no_check` supports all of `change`'s variants.
+
+   .. tacv:: change_no_check @term with @term’
+      :undocumented:
+
+   .. tacv:: change_no_check @term at {+ @num} with @term’
```

Comment:
Does the `at n` modifier really come *before* the `with` modifier?  That seems backwards...

Reply:
Honestly, I started by copy-pasting the existing docs for `change`
https://github.com/coq/coq/blob/6960da4736186fa6214854329f36f558e7aa4d0b/doc/sphinx/proof-engine/tactics.rst#L2932, assuming they're correct. Are they?

Both the source and the PR comments suggest `change_no_check` has the same options as `change`, so adapting those docs should be correct.

Reply:
Actually both orders are supported, but I agree with @JasonGross that the only natural order is the one where `with` comes before `at`.

Reply:
In fact, this order makes sense too: it is the first term, which is "at" a position...